x/staking/keeper/keeper.go

```diff
@@ -77,6 +78,7 @@ func NewKeeper(
 		validatorAddressCodec: validatorAddressCodec,
 		consensusAddressCodec: consensusAddressCodec,
 		LastTotalPower:        collections.NewItem(sb, types.LastTotalPowerKey, "last_total_power", sdk.IntValue),
+		HistoricalInfo:        collections.NewMap(sb, types.HistoricalInfoKey, "historical_info", collections.Int64Key, codec.CollValue[types.HistoricalInfo](cdc)),
```

Comment:
Are we sure int64Key of collections matches the same bytes representation as:
```
https://github.com/cosmos/cosmos-sdk/blob/main/x/staking/types/keys.go#L423C19-L423C19
```

Reply:
Yeah, made some changes to work for legacy key, https://github.com/cosmos/cosmos-sdk/pull/17063/commits/b6df8d5fe379e2184fc5afff2cf2f3b4eddcd6f1